src/components/Modal/RegisterModal.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from 'react';
+import { emailPattern } from '../../utils/constant';
+import Button from '../Button/Button';
+import Modal from './Modal';
+import './RegisterModal.scss';
+
+const RegisterModal = ({ onClose, handleRedirect }) => {
+  const [userInfo, setUserInfo] = useState({
+    name: '',
+    email: '',
+    password: '',
+  });
+
+  const { name, email, password } = userInfo;
+
+  const emailIsValid = emailPattern.test(email);
+  const passwordIsValid = password.length >= 5;
+  const isValid = emailIsValid && passwordIsValid;
+  const isDisabled = isValid && name !== '';
+
+  const handleInputChange = e => {
+    const { value, id } = e.target;
+    setUserInfo(prev => ({
+      ...prev,
+      [id]: value,
+    }));
+  };
+
+  const handleRegister = e => {
+    if (isValid) {
+      fetch('http://10.58.52.81:3000/user/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify(userInfo),
+      })
+        .then(res => res.json())
+        .then(data => console.log(data));
+    }
+  };
```

Comment:
- e.preventDefault() 함수를 실행해야 form 태그의 submit 동작 시의 새로고침을 막을 수 있겠네요!
- 그리고 handleRegister의 동작 자체를 if문 안에 넣는 것보다는, 예외 케이스를 미리 제거한 다음 실제 동작인 fetch를 작성하는 방식이 일반적으로 많이 사용되는 early return 방식입니다. 코드 첨부해드릴 테니 참고해 보세요!
```suggestion
  const handleRegister = e => {
    e.preventDefault();
    
    if (!isValid) {
      alert('입력 값을 확인해 주세요!');
      
      return;
    }
    
    fetch('http://10.58.52.81:3000/user/signup', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json;charset=utf-8',
      },
      body: JSON.stringify(userInfo),
    })
      .then(res => res.json())
      .then(data => console.log(data));
  };
```